editor/editor_inspector.cpp

```diff
@@ -1979,7 +1978,10 @@ void EditorInspector::_property_changed(const String &p_path, const Variant &p_v
 	if (changing)
 		this->changing++;
 
-	_edit_set(p_path, p_value, false, p_name);
+	if (p_path == "script")
+		_edit_set(p_path, p_value, true, p_name);
+	else
+		_edit_set(p_path, p_value, false, p_name);
```

Comment:
If the only change between the `if` branches is the boolean argument, I think it's clearer to write as `_edit_set(p_path, p_value, p_path == "script", p_name);`